File: dotnet/EmitLogTopic.cs
using System;
using System.Linq;
using RabbitMQ.Client;
using System.Text;

class EmitLogTopic
{
    public static void Main(string[] args)
    {
        var factory = new ConnectionFactory() { HostName = "localhost" };
        using (var connection = factory.CreateConnection())
        {
            using (var channel = connection.CreateModel())
            {
                channel.ExchangeDeclare("topic_logs", "topic");

                var routingKey = (args.Length > 0) ? args[0] : "anonymous.info";
                var message = (args.Length > 1) ? string.Join(" ", args.Skip(1)
                                                                                  .ToArray())
                                                               : "Hello World!";
                var body = Encoding.UTF8.GetBytes(message);
                channel.BasicPublish("topic_logs", routingKey, null, body);
                Console.WriteLine(" [x] Sent '{0}':'{1}'", routingKey, message);
            }
        }
    }
}

File: julia/emit_log_topic.jl
using AMQPClient
const VIRTUALHOST = "/"
const HOST = "127.0.0.1"

function send()
    # 1. Create a connection to the localhost or 127.0.0.1 of virtualhost '/'
    connection(; virtualhost=VIRTUALHOST, host=HOST) do conn
        # 2. Create a channel to send messages
        channel(conn, AMQPClient.UNUSED_CHANNEL, true) do chan
            # 3. Declare Exchange
            exchange = "topic_logs"
            exchange_topic = "topic"
            exchange_declare(chan, exchange, EXCHANGE_TYPE_TOPIC)
            # 4. Get input data
            if length(Base.ARGS) >= 2
                routing_key = Base.ARGS[1]
                received = join(Base.ARGS[2:end], ' ')
            else
                routing_key = "info"
                received = "Hello World"
            end
            # 5. Prepare and send data
            data = convert(Vector{UInt8}, codeunits(received))
            msg = Message(data, content_type="text/plain", delivery_mode=PERSISTENT)
            basic_publish(chan, msg; exchange=exchange, routing_key=routing_key)
            println("Message sent: $received, routing key: $routing_key")
        end
    end
end

send()
